### DISCUSSION TITLE
ffmpeg fix for clang

### DISCUSSION THREAD

**Discussion by revelator** [msys2/MINGW-packages#9258] [General]
patches out inlining for clang on x86 in cabac.h allowing it to build.

[0004-clang-disable-cabac-x86-inlining.zip](https://github.com/msys2/MINGW-packages/files/6912287/0004-clang-disable-cabac-x86-inlining.zip)

**Comment by 1480c1** [comment]
I guess I never sent out the patch https://raw.githubusercontent.com/m-ab-s/mabs-patches/master/ffmpeg/0001-get_cabac_inline_x86-Don-t-inline-if-32-bit-clang-on.patch

@mstorsjo I noticed you had a patch a long time ago relating to clang+32-bit on windows https://patchwork.ffmpeg.org/project/ffmpeg/patch/20200523182917.11853-1-martin@martin.st/ any comment regarding that?

**Reply by mstorsjo** [reply]
I had a plan of getting back to that patch, but I never got to it. This patch probably is better than mine, as clang can build that assembly just fine, it just fails to fulfill all the assembly constraints when inlined into a larger context.

**Reply by revelator** [reply]
also seems to have been a problem on android from what i could gather from the bug reports, cabac seems to be rather picky in that regard. Found this patch while looking for fixes for windows clang builds though i believe it was originally intended for msvc clang it also worked for ours.

**Reply by 1480c1** [reply]
https://trac.ffmpeg.org/ticket/8903 found the old ticket I made for this, it seems I attached the patch there but forgot to ever git send-email it

**Comment by 1480c1** [comment]
https://patchwork.ffmpeg.org/project/ffmpeg/patch/20210817153539.188484-1-ccom@randomderp.com/

sent upstream

**Reply by 1480c1** [reply]
@mstorsjo do you know what would be the best way to benchmark this function? I'm not sure if decoding any h264/h265 video will use this function

**Reply by mstorsjo** [reply]
Most H264 videos would exercise it (in practice, main/high profile content, unless the file was explicitly encoded with CAVLC enabled instead of CABAC), and afaik all HEVC videos. And higher bitrate content exercises it more than lower bitrate.

But I nerdsniped myself into benchmarking it already, see the mailing list.

**Reply by 1480c1** [reply]
Thanks nerdsniper, interesting results btw

**Reply by mstorsjo** [reply]
I pushed this patch now, see https://github.com/FFmpeg/FFmpeg/commit/8990c5869e27fcd43b53045f87ba251f42e7d293.
